x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/OptimizerRulesTests.java

```diff
@@ -138,6 +138,6 @@ protected Expression rule(Expression e, LogicalOptimizerContext ctx) {
         var alias = new Alias(new Source(1, 18, "x=f1+1"), "x", add);
 
         // contains expressions only from EVAL
-        assertThat(rule.appliedTo, contains(alias, add, attribute, literal));
+        assertThat(rule.appliedTo, containsInAnyOrder(alias, add, attribute, literal));
```

Comment:
This might appear as `alias, add, attribute, literal` or `attribute, literal, add, alias` depending on `TransformDirection`